### PR TITLE
Promote orphan pod adoption E2E test for Jobs to conformance

### PR DESCRIPTION
**Pull request by globervinodhn**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind api-change
> /kind bug
/kind cleanup
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

**What this PR does / why we need it**: This PR request for the promotion of an existing E2E test for Jobs to conformance. In this test, the job is tested if it adopts an orphaned pod and removes the non-matching pods.

**Which issue(s) this PR fixes**: Request to promote an existing E2E to conformance
Fixes #

**Special notes for your reviewer**:
1. The existing E2E takes approximately 55 seconds for successful completion.
2. The test is found to be non-disruptive and non-flaky in nature.
3. The test verifies an important feature wherein Jobs can adopt an orphaned pod matching the specified specification.

**Does this PR introduce a user-facing change?**: NONE

release-note-none

cc @brahmaroutu @mgdevstack

### CODE REVIEW

File: test/e2e/apps/job.go [hunk 1]
```diff
@@ -118,7 +118,12 @@ var _ = SIGDescribe("Job", func() {
 		Expect(errors.IsNotFound(err)).To(BeTrue())
 	})
 
-	It("should adopt matching orphans and release non-matching pods", func() {
+	/*
+		Release : v1.15
+		Testname: Jobs, orphan pods, re-adoption
+		Description: Create a job with paralellism equal to 2. Ensure to make one of the pod orphan by modifying the pod's owner's reference value. Check if the Job re-adopts the orphan pods. Also, ensure that the non-matching pods are released
```
Comment (neolit123) [inline test/e2e/apps/job.go]: > pod's owner's reference value
->
> pod owner reference value

also please split the description on multiple lines.

**Review by neolit123 — COMMENTED** [review]
/release-note-none
/ok-to-test
/priority important-longterm
@kubernetes/sig-apps-pr-reviews

File: test/e2e/apps/job.go (hunk 1 above)
Comment (globervinodhn) [reply test/e2e/apps/job.go]: Thanks for the review. The changes have been done as requested

File: test/conformance/testdata/conformance.txt [hunk 2]
```diff
@@ -24,13 +24,8 @@ test/e2e/apps/deployment.go: "RecreateDeployment should delete old pods and crea
 test/e2e/apps/deployment.go: "deployment should delete old replica sets"
 test/e2e/apps/deployment.go: "deployment should support rollover"
 test/e2e/apps/deployment.go: "deployment should support proportional scaling"
-test/e2e/apps/job.go: "should delete a job"
```
Comment (mgdevstack) [inline test/conformance/testdata/conformance.txt]: Are they deleted by mistake? Please review locally before making a final push.

File: test/e2e/apps/job.go [hunk 3]
```diff
@@ -124,8 +124,20 @@ var _ = SIGDescribe("Job", func() {
 		gomega.Expect(errors.IsNotFound(err)).To(gomega.BeTrue())
 	})
 
+<<<<<<< HEAD
```
Comment (mgdevstack) [inline test/e2e/apps/job.go]: Please resolve the conflicts appropriately.

**Comment by globervinodhn** [comment]
PR waiting for review @brahmaroutu

File: test/e2e/apps/job.go [hunk 4]
```diff
@@ -132,7 +132,14 @@ var _ = SIGDescribe("Job", func() {
...
 	})
 
-	ginkgo.It("should adopt matching orphans and release non-matching pods", func() {
+	/*
+		Release : v1.16
+		Testname: Jobs, orphan pods, re-adoption
+		Description: Create a job with paralellism equal to 2. Ensure to make one of the pod orphan by
+		modifying the pod's owner's reference value. Check if the Job re-adopts the orphan pods.
+		Also, ensure that the non-matching pods are released. The test MUST pass
+	*/
```
Comment (johnbelamaric) [inline test/e2e/apps/job.go]: Let's change the description to be:

```
Create a parallel job. The number of Pods MUST equal the level of parallelism. Orphan a Pod by modifying its owner reference. The Job MUST re-adopt the orphan pod. Modify the labels of one of the Job's Pods. The Job MUST release the Pod.
```

Comment (globervinodhn) [reply test/e2e/apps/job.go]: The changes have been committed :-)

Comment (globervinodhn) [reply test/e2e/apps/job.go]: The description has been updated and committed.

**Comment by spiffxp** [comment]
/lgtm
behavior seems sane, and is written well enough

**Comment by spiffxp** [comment]
/close
closing in favor of https://github.com/kubernetes/kubernetes/pull/80544

**Pull request closed without merging**